dask/dataframe/shuffle.py

```diff
@@ -624,7 +648,10 @@ def shuffle_group_3(df, col, npartitions, p):
 
 
 def set_index_post_scalar(df, index_name, drop, column_dtype):
-    df2 = df.drop("_partitions", axis=1).set_index(index_name, drop=drop)
+    if not drop:
```

Comment:
Why this?  cuDF doesn't support this keyword?  Could we add it there rather than here?

Reply:
Right, cudf currently requires you to drop the column when you set it as an index.  It would certainly be ideal to add thefeature to cudf.  However, this PR was intended as a one-stop temporary solution for cudf support.

Reply:
Right, we're probably not going to accept a temporary solution like this in the dask codebase.  

This project currently optimizes heavily for the main userbase, which is Pandas based, not cudf.

Reply:
Totally understand - This is the reason for the `[WIP]` label (it will be removed when I belive this is a long-term solution, etc)

Reply:
Note: [cudf#2489](https://github.com/rapidsai/cudf/pull/2489/files) should resolve this soon.